src/commands/createFunction/FunctionListStep.ts

```diff
@@ -50,7 +50,8 @@ export class FunctionListStep extends AzureWizardPromptStep<IFunctionWizardConte
                     const actualId: string = t.id.toLowerCase();
                     const expectedId: string = options.templateId.toLowerCase();
                     // Check for a match with or without the language in the id
-                    return actualId === expectedId || actualId === `${expectedId}-${language.toLowerCase()}`;
+                    // The id for C# triggers include "CSharp" rather than "C#", so we'll have to search for that instead
+                    return actualId === expectedId || actualId.includes(expectedId) && actualId.includes((language === ProjectLanguage.CSharp ? 'CSharp' : language).toLowerCase());
```

Comment:
The "includes" logic is too simple. For example, there's an "HttpTrigger" template and an "HttpTriggerWithParameters" template that would cause problems.

Also - we don't have to check for the language in particular. It's already filtered by language in the call to "getFunctionTemplates" a few lines up. The only reason I used language before was because I was doing `===` instead of `includes`.